packages/plugins/RedPacket/src/SiteAdaptor/RedPacket/index.tsx

```diff
@@ -279,7 +279,8 @@ export const RedPacket = memo(function RedPacket({ payload }: RedPacketProps) {
                 return
             }
             hash = await claimCallback()
-            if (platform && myProfileId && myHandle && hash) {
+            // `hash` could be an `Error` in runtime
```

Comment:
why you don't check like `hash instanceof Error`?

Reply:
I don't care about any other types but only string. `hash` must be a string.

The `useAsyncFn` behind `claimCallback` returns error instead of throwing it back unexpectedly. It makes the runtime unpredictable.


https://github.com/streamich/react-use/blob/ade8d3905f544305515d010737b4ae604cc51024/src/useAsyncFn.ts#L58-L61

Reply:
If hash could be Error in runtime, the type of it should be fixed.

Reply:
No. Already tested.

The error has been caught and handled by return it.

You might take a look at https://github.com/streamich/react-use/blob/ade8d3905f544305515d010737b4ae604cc51024/src/useAsyncFn.ts#L58-L61

Reply:
Let's create a custom hook to fix this.